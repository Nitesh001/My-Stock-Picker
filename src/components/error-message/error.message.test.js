import React from 'react';
import renderer from 'react-test-renderer';
import ErrorMessage from './error.message';

test('Error component shows default message', () => {
  const component = renderer.create(
    <ErrorMessage />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  // tree.props.onClose();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onMouseLeave();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});

test('Error component shows custom message', () => {
  const component = renderer.create(
    <ErrorMessage mssg="custom error" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Error component with onclose', () => {
  const component = renderer.create(
    <ErrorMessage mssg="custom error" onClose={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})