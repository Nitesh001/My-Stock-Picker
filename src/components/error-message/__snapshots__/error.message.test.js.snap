// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component shows custom message 1`] = `
<div
  className="fade alert alert-info alert-dismissible show"
  role="alert"
>
  <button
    aria-label="Close alert"
    className="btn-close"
    onClick={[Function]}
    type="button"
  />
  <p>
    custom error
  </p>
</div>
`;

exports[`Error component shows default message 1`] = `
<div
  className="fade alert alert-info alert-dismissible show"
  role="alert"
>
  <button
    aria-label="Close alert"
    className="btn-close"
    onClick={[Function]}
    type="button"
  />
  <p>
    Sorry! Please try again. We don't have any record matching your criteria
  </p>
</div>
`;

exports[`Error component with onclose 1`] = `
<div
  className="fade alert alert-info alert-dismissible show"
  role="alert"
>
  <button
    aria-label="Close alert"
    className="btn-close"
    onClick={[Function]}
    type="button"
  />
  <p>
    custom error
  </p>
</div>
`;
